_posts/2019-08-22-view-encapsulation-w-angularze.md

```diff
@@ -0,0 +1,445 @@
+
+W przykładzie przenieśliśmy komponent `app-green` z komponentu `app-root` do komponentu `app-blue`, usunęliśmy jego style i ustawiliśmy domyślny tryb kapsułkowania.  
+[Link do repozytorium](https://github.com/Michuu93/view-encapsulation-demo/tree/ViewEncapsulation.ShadowDom)
+
+```typescript
+import {Component, ViewEncapsulation} from '@angular/core';
+
+@Component({
+    selector: 'app-red',
+    template: `
+        <p>Red paragraph!</p>
+    `,
+    styles: [`
+        p {
+            color: red;
+        }
+    `],
+    encapsulation: ViewEncapsulation.ShadowDom
+})
+export class RedComponent {
+}
+```
+
+```typescript
+import {Component} from '@angular/core';
+
+@Component({
+    selector: 'app-green',
+    template: `
+        <p>Green paragraph!</p>
+    `
+})
+export class GreenComponent {
+}
+```
+
+```typescript
+import {Component, ViewEncapsulation} from '@angular/core';
+
+@Component({
+    selector: 'app-blue',
+    template: `
+        <app-green></app-green>
+        <p>Blue paragraph!</p>
+    `,
+    styles: [`
+        p {
+            color: blue;
+        }
+    `],
+    encapsulation: ViewEncapsulation.ShadowDom
+})
+export class BlueComponent {
+}
+```
+
+Wynikowy kod HTML:
+
+```html
+<head>
+</head>
+
+<body>
+    <app-root ng-version="8.2.2">
+        <app-red>
+            #shadow-root
+            <style>
+                p {
+                    color: red;
+                }
+            </style>
+            <p>Red paragraph!</p>
+        </app-red>
+        <app-blue>
+            #shadow-root
+            <style>
+                p {
+                    color: blue;
+                }
+            </style>
+            <app-green>
+                <p>Green paragraph!</p>
+            </app-green>
+            <p>Blue paragraph!</p>
+        </app-blue>
+    </app-root>
+</body>
+```
+
+![ViewEncapsulation.ShadowDom](/assets/img/posts/2019-08-22-view-encapsulation/view_encapsulation_shadow_dom_result.jpg)
+<span class="img-legend">Wynik widoczny w przeglądarce</span>
+
+Tryb Shadow DOM również pozwolił nam odseparować style między poszczególnymi komponentami. W sekcji `<head>` nie ma już żadnych stylów, natomiast są ukryte w Shadow Root elementów DOM-u. Na przykładzie widzimy, że style z komponentu rodzica `app-blue` zostały zaaplikowane do komponentu dziecka `app-green`, w efekcie czego paragraf ma kolor niebieski. Gdyby komponent `app-green` również posiadał tryb `ViewEncapsulation.ShadowDom`, to style rodzica nie zostałyby zaaplikowane, ponieważ korzystałby ze stylów z własnego Shadow Root.
+
+### ~~ViewEncapsulation.Native~~
+
+Do niedawna zamiast `ViewEncapsulation.ShadowDom` dostępny był tryb `ViewEncapsulation.Native`.  
+Działał on w podobny sposób, ale został wycofany z powodu wykorzystywania przestarzałego standardu Shadow DOM.
+
+---
```

Comment:
Ta linia jest zbędna i wygląda słabo... IMO to zdanie podsumowania spokojnie może być w oddzielnym paragrafie bez żadnego dodatkowego graficznego wyróżnienia. Ew. można dodać nagłówek "Podsumowanie"